src/js/sharedEth/utils.js

```diff
@@ -61,6 +61,26 @@ function providerType(provider) {
   }
 }
 
+function providerTypeId(provider) {
+  if (provider === undefined || provider == null) {
+    return 3;
+  } else if (provider.isToshi) {
+    return 3;
+  } else if (provider.isImToken) {
+    return 3;
+  }else if (provider.isTally) {
+    return 5;
+  } else if (provider.isMetaMask) {
+    if (provider.constructor.name === 'InpageBridge') {
+      return 3;
+    } else {
+      return 3;
+    }
+  } else {
+    return null;
```

Comment:
Here we probably want to return 3 instead of null? would not recommend to return a non number from this function lest something that uses it may crash.